### DISCUSSION TITLE
vcpkg does not consider supported architectures when selecting a Visual Studio instance

### DISCUSSION THREAD

**Discussion by thespivey** [microsoft/vcpkg#19013] [Ideas]
On machines that have multiple versions of Visual Studio installed (Community + Enterprise), with x86/x64 tools in both but ARM tools only installed in the Enterprise instance, vcpkg will select the Community edition for an arm build and then fail.

```
-- Running vcpkg install
Detecting compiler hash for triplet x64-windows...
Detecting compiler hash for triplet arm-uwp...
Error: Unsupported toolchain combination.
Target was arm but the chosen Visual Studio instance supports:
    x86, amd64, x86_amd64, amd64_x86
Vcpkg selected C:\Program Files (x86)\Microsoft Visual Studio\2019\Community as the Visual Studio instance.
Detected instances:
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise
```    

Removing the x86/x64 tools from the Community install will cause vcpkg to select the Enterprise edition instead and build successfully.

**Comment by JackBoosY** [comment]
@vicroms what do you think about?

**Comment by vicroms** [comment]
@thespivey it is possible to tell vcpkg which Visual Studio instance to use by setting [`VCPKG_VISUAL_STUDIO_PATH`](https://vcpkg.io/en/docs/users/triplets.html#VCPKG_VISUAL_STUDIO_PATH) as an environment variable or as a variable in a custom triplet file.

**Reply by thespivey** [reply]
Thanks.  Yeah, our fallback if vcpkg can't do it is to wrap the build with other tooling (something using vswhere probably) that can select the right instance and bypass all the code in vcpkg that autoselects.
